src/extension/sldService/src/main/java/org/geoserver/sldservice/rest/ClassifierController.java

```diff
@@ -453,6 +465,33 @@ private JSONObject jsonRule(Object obj) {
         return Collections.singletonList(rule);
     }
 
+    private void adaptSingleValueColorMap(RenderedImage image, ColorMap colorMap) {
+        // force it to be visible, depending on the method the first entry might be
+        // transparent
+        ColorMapEntry cm0 = colorMap.getColorMapEntry(0);
+        cm0.setOpacity(FF.literal(1));
+
+        // should always be a literal, just covering for possible future changes
+        if (cm0.getQuantity() instanceof Literal) {
+            // wrap it between two values that are slightly below and above
+            Float value = cm0.getQuantity().evaluate(null, Float.class);
+            int intBits = Float.floatToIntBits(value);
+            ColorMapEntry cm1 = new ColorMapEntryImpl();
+            cm1.setQuantity(FF.literal(Float.intBitsToFloat(intBits + 1)));
+            cm0.setQuantity(FF.literal(Float.intBitsToFloat(intBits - 1)));
+
+            cm1.setColor(cm0.getColor());
+            cm1.setLabel(cm0.getLabel());
+            cm1.setOpacity(cm0.getOpacity());
+
+            colorMap.addColorMapEntry(cm1);
+            colorMap.setType(ColorMap.TYPE_INTERVALS);
+        } else {
+            // make it formally valid, but the value match will not really not work
```

Comment:
double negate? :) (will not not work -> will work :P)

Reply:
LOL, fixing it.